rotkehlchen/tasks/events.py

```diff
@@ -86,45 +86,65 @@ def match_asset_movements(database: 'DBHandler') -> None:
+            events_db=events_db,
+            asset_movement=asset_movement,
+            matched_event=matched_event,
+            is_deposit=is_deposit,
+        )
+
+
+def update_asset_movement_matched_event(
+        events_db: DBHistoryEvents,
+        asset_movement: AssetMovement,
+        matched_event: HistoryBaseEntry,
+        is_deposit: bool,
+) -> tuple[bool, str]:
+    """Update the given matched event with proper event_type, counterparty, etc and cache the
+    event identifiers. Returns a tuple containing a boolean indicating success and a string
+    containing any error message.
+    """
+    if isinstance(matched_event, OnchainEvent):
         # This could also be a plain history event (i.e. a btc event, or custom event)
         # so only set counterparty if the event supports it.
-        if isinstance(matched_event, OnchainEvent):
-            if matched_event.counterparty is not None:
-                # This event has been decoded by a protocol decoder and likely isn't the match
-                # TODO: Figure out how to handle some cases like swaps and bridging where a
-                # protocol event could be the corresponding event for an asset movement.
-                continue
-
-            matched_event.counterparty = asset_movement.location.name.lower()
-
-        # Modify the matched event
-        if is_deposit:
-            matched_event.event_type = HistoryEventType.DEPOSIT
-            matched_event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
-            notes = 'Deposit {amount} {asset} to {exchange}'
-        else:
-            matched_event.event_type = HistoryEventType.WITHDRAWAL
-            matched_event.event_subtype = HistoryEventSubType.REMOVE_ASSET
-            notes = 'Withdraw {amount} {asset} from {exchange}'
-
-        matched_event.notes = notes.format(
-            amount=matched_event.amount,
-            asset=matched_event.asset.resolve_to_asset_with_symbol().symbol,
-            exchange=asset_movement.location_label,
+        if matched_event.counterparty is not None:
+            # This event has been decoded by a protocol decoder and likely isn't the match
+            # TODO: Figure out how to handle some cases like swaps and bridging where a
+            # protocol event could be the corresponding event for an asset movement.
+            return False, 'Event already has a counterparty set'
+
+        matched_event.counterparty = asset_movement.location.name.lower()
+
+    # Modify the matched event
+    if is_deposit:
+        matched_event.event_type = HistoryEventType.DEPOSIT
+        matched_event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+        notes = 'Deposit {amount} {asset} to {exchange}'
+    else:
+        matched_event.event_type = HistoryEventType.WITHDRAWAL
+        matched_event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+        notes = 'Withdraw {amount} {asset} from {exchange}'
+
+    matched_event.notes = notes.format(
```

Comment:
I would save some of this data in extra data so we can match them better. The from/to I mean. We need to do exactly the same thing with bridging